xl-v1.go

```diff
@@ -429,8 +429,26 @@ func (xl XL) StatFile(volume, path string) (FileInfo, error) {
 		return FileInfo{}, errInvalidArgument
 	}
 
+	readLock := true
```

Comment:
> The readLock variable is to differentiate if its a write lock or readlock operation.

sure, what I meant is to simply a little the API, so it would be:
  xl.lockNS(volume, path, true/false)
  xl.unlockNS(volume, path)